Gems/Atom/Feature/Common/Assets/Shaders/Materials/EnhancedPBR/EnhancedPBR_VertexEval.azsli

```diff
@@ -24,10 +24,11 @@ VsOutput EvaluateVertexGeometry_EnhancedPBR(
     float4 tangent,
     float2 uv0,
     float2 uv1,
-    uint instanceId)
+    uint instanceId,
+    float4 color0)
 {
     // BasePBRP
-    VsOutput output = EvaluateVertexGeometry_BasePBR(position, normal, tangent, uv0, uv1, instanceId);
+    VsOutput output = EvaluateVertexGeometry_BasePBR(position, normal, tangent, uv0, uv1, color0, instanceId, o_color0_isBound);
```

Comment:
o_color0_isBound is defined behind ENABLE_VERTEX_COLOR macro so can we even use it here. Does the shader compile fine?

Reply:
I am wondering can we ever be in a scenario (i.e render pipeline) whereby we have ENABLE_VERTEX_COLOR disabled for EnhancedPBR. To me this can happen (in theory) and if we need to do something like.

bool useVertexColor0 = true; //or false?
#if ENABLE_VERTEX_COLOR
    useVertexColor0 = o_color0_isBound;
 #endif
// pass useVertexColor0 to EvaluateVertexGeometry_BasePBR